src/git-source-provider.ts

```diff
@@ -57,6 +57,16 @@ export async function getSource(settings: IGitSourceSettings): Promise<void> {
     core.info(
       `To create a local Git repository instead, add Git ${gitCommandManager.MinimumGitVersion} or higher to the PATH`
     )
+    if (settings.submodules) {
```

Comment:
fail when fallback to REST API and submodules or ssh-key is set

Reply:
Shouldn't we do this before logging 'The repository will be downloaded using the GitHub REST API'

Since errors propagate up as annotations. Consider: "Git ${gitCommandManager.MinimumGitVersion} version or greater not found in path. Input 'submodules' not supported when falling back to download using the GitHub REST API"

Same for ssh key

Reply:
I really like failing as an error flow here 👍

Reply:
Updated. To make the error actionable. I switched the order a little bit from the suggestion, to lead with the error message followed by suggested action.